Add my project to dev

OleksandrHolopapa opened the pull request:
My project. jar-file is in Releases

HolopapaOleksandr committed:
Add my project to dev

HolopapaOleksandr committed:
- changes in the names of some variables and methods
- improvement of output of messages using Exceptions
- adding the LENGTH_OF_ALPHABET constant to the AlgorithmOfCaesar class
- command, sourceFilePath, key made final
- changes in the ReadMe file

HolopapaOleksandr committed:
changes in the ReadMe. Add jar-file address

HolopapaOleksandr committed:
- redesigned CLI class to use the swing library
- the main method starts the program window
- three new variables and three getters were added in the FileWork class
- the getKey() method in the FileWork class has changed
- changed the Runner class to interact with the CLI class

HolopapaOleksandr committed:
- changes in the readAndWrite() method of the FileWork class
- changes in the run method of the Runner class
- change of executeCommandButton logic
- change ReadMe

HolopapaOleksandr committed:
set private getCommandIndex()

HolopapaOleksandr committed:
add release address

HolopapaOleksandr committed:
changes in CLI class

HolopapaOleksandr committed:
change address of release in ReadMe.txt

HolopapaOleksandr committed:
- changed FileWork class logic
- some changers in methods and vars names in other classes
- change in ReadMe

HolopapaOleksandr committed:
add Exception colling to method findKeyAndInitializeKeyAndFinalFilePathAndFinalTextForBruteForce of FileWork class
some final settings

HolopapaOleksandr committed:
change methods names in FileWork, AlgorithmOfCaesar, CLI classes

HolopapaOleksandr committed:
ReadMe

HolopapaOleksandr committed:
Some changes in FileWork class

nemovruslan commented on ReadMe.txt:
```diff
@@ -0,0 +1,63 @@
+ВИКОРИСТОВУВАВ: openjdk-23.
```
Змініть будь ласка розширення для файлу з ReadMe.txt на ReadMe.md, після цього зявиться функціонал який допоможе використовувати функції для форматування тексту.

nemovruslan commented on src/Runner.java:
```diff
@@ -0,0 +1,26 @@
+        } catch (IOException e) {
+            frame.showMassage("Помилка читання/запису файлу. Перевірте правильність шляху до файлу");
+        }
+
```
Видаліть пустий рядок

nemovruslan commented on src/FileWork.java:
```diff
@@ -0,0 +1,80 @@
+            }
+        }
+        if (key.equals("0")) throw new RuntimeException("Не вдалося підібрати ключ");
+
```
Видаліть пустий рядок

nemovruslan commented on src/FileWork.java:
```diff
@@ -0,0 +1,80 @@
+            case "BRUTE_FORCE" -> bruteForce(algorithm);
+        }
+    }
+
```
Видаліть пустий рядок

nemovruslan commented on src/CLI.java:
```diff
@@ -0,0 +1,146 @@
+
+    CLI(String[] args) {
+        this();
+        if (isCommandCorrect(args[0])) commandsComboBox.setSelectedIndex(getCommandIndex(args[0]));
```
Писати бізнес логіку у конструкторі погана практика. Будь ласка винесіть бізнес логіку у методи.

nemovruslan commented on src/CLI.java:
```diff
@@ -0,0 +1,146 @@
+        if (args.length > 2) keyValueTextField.setText(args[2]);
+    }
+
+    CLI() {
```
Занадто великий конструктор, краще винесіть дану логіку у метод.

nemovruslan commented on src/CLI.java:
```diff
@@ -0,0 +1,146 @@
+            if (Objects.requireNonNull(commandsComboBox.getSelectedItem()).toString().equals("BRUTE FORCE")) {
+                keyValueTextField.setText(runner.getFileWork().getKey());
+            }
+        } catch (NullPointerException _) {
```
Якщо Ви не обробляєте помилку то краще замість _ назвати ignored.

nemovruslan commented on src/Runner.java:
```diff
@@ -0,0 +1,26 @@
+import java.io.IOException;
+
+class Runner {
```
У пакеті src рекомендується створювати пакети відповідно до компанії у якій створюється даний продукт. Наприклад якщо назва компанії javarush.com, то вкладеність пакетів буде навиворіт com.javarush, де javarush буде лежати у пакеті com, а у пакеті javarush  будуть лежати ваші класи.

HolopapaOleksandr committed:
-moving all classes to package com.javarush
-add new methods to CLI
-delete method isCommandCorrect from CLI class
-new file ReadMe.md
-correction of comments
